server/functions/framework/compiled_function.go

```diff
@@ -88,22 +92,28 @@ func newCompiledFunctionInternal(
 	c.callResolved = make([]pgtypes.DoltgresType, len(functionParameterTypes)+1)
 	hasPolymorphicParam := false
 	for i, param := range functionParameterTypes {
-		if _, ok := param.(pgtypes.DoltgresPolymorphicType); ok {
+		if param.IsPolymorphicType() {
 			// resolve will ensure that the parameter types are valid, so we can just assign them here
 			hasPolymorphicParam = true
 			c.callResolved[i] = originalTypes[i]
 		} else {
+			if d, ok := args[i].Type().(pgtypes.DoltgresType); ok {
+				// `param` is a default type which does not have type modifier set
+				param.AttTypMod = d.AttTypMod
+			}
 			c.callResolved[i] = param
 		}
 	}
 	returnType := fn.GetReturn()
 	c.callResolved[len(c.callResolved)-1] = returnType
-	if _, ok := returnType.(pgtypes.DoltgresPolymorphicType); ok {
+	if returnType.IsPolymorphicType() {
 		if hasPolymorphicParam {
 			c.callResolved[len(c.callResolved)-1] = c.resolvePolymorphicReturnType(functionParameterTypes, originalTypes, returnType)
+		} else if c.Name == "array_in" || c.Name == "array_recv" {
+			// TODO: `array_in` and `array_recv` functions don't follow this rule
+			// The return type should resolve to the type of OID value passed in as second argument.
```

Comment:
This doesn't seem quite right. I'll take a look at this once you've merged this in. In which test did you notice this misbehaving?

Reply:
All array type tests from `types_test.go` fails on this.